modules/evm/src/tests.rs

```diff
@@ -491,12 +511,66 @@ fn should_transfer_from_contract() {
 
 		assert_eq!(result.exit_reason, ExitReason::Succeed(ExitSucceed::Stopped));
 		assert_eq!(balance(charlie()), 3 * amount);
-		assert_eq!(balance(alice()), INITIAL_BALANCE - 3 * amount);
+		assert_eq!(
+			balance(alice()),
+			INITIAL_BALANCE - 3 * amount - <Test as Trait>::ContractExistentialDeposit::get()
+		);
 	})
 }
 
 #[test]
 fn contract_should_deploy_contracts() {
+	// pragma solidity ^0.5.0;
+	//
+	// contract Factory {
+	//     Contract[] newContracts;
+	//
+	//     function createContract () public payable {
+	//         Contract newContract = new Contract();
+	//         newContracts.push(newContract);
+	//     }
+	// }
+	//
+	// contract Contract {}
+	let contract = from_hex("0x608060405234801561001057600080fd5b5061016f806100206000396000f3fe608060405260043610610041576000357c0100000000000000000000000000000000000000000000000000000000900463ffffffff168063412a5a6d14610046575b600080fd5b61004e610050565b005b600061005a6100e2565b604051809103906000f080158015610076573d6000803e3d6000fd5b50905060008190806001815401808255809150509060018203906000526020600020016000909192909190916101000a81548173ffffffffffffffffffffffffffffffffffffffff021916908373ffffffffffffffffffffffffffffffffffffffff1602179055505050565b6040516052806100f28339019056fe6080604052348015600f57600080fd5b50603580601d6000396000f3fe6080604052600080fdfea165627a7a7230582092dc1966a8880ddf11e067f9dd56a632c11a78a4afd4a9f05924d427367958cc0029a165627a7a723058202b2cc7384e11c452cdbf39b68dada2d5e10a632cc0174a354b8b8c83237e28a40029").unwrap();
+	new_test_ext().execute_with(|| {
+		let result = <Test as Trait>::Runner::create(alice(), contract.clone(), 0, 1000000000).unwrap();
+		assert_eq!(result.exit_reason, ExitReason::Succeed(ExitSucceed::Returned));
+		assert_eq!(
+			balance(alice()),
+			INITIAL_BALANCE - <Test as Trait>::ContractExistentialDeposit::get()
+		);
+		let factory_contract_address = result.address;
+		assert_eq!(balance(result.address), 0);
+		assert_eq!(
+			reserved_balance(result.address),
+			<Test as Trait>::ContractExistentialDeposit::get()
+		);
+
+		// Factory.createContract
+		let amount = 1000000000;
+		let create_contract = from_hex("0x412a5a6d").unwrap();
+		let result =
+			<Test as Trait>::Runner::call(alice(), result.address, create_contract, amount, 1000000000).unwrap();
+		assert_eq!(result.exit_reason, ExitReason::Succeed(ExitSucceed::Stopped));
+		assert_eq!(
+			balance(alice()),
+			INITIAL_BALANCE - amount - <Test as Trait>::ContractExistentialDeposit::get()
+		);
+		assert_eq!(
+			balance(factory_contract_address),
+			amount - <Test as Trait>::ContractExistentialDeposit::get()
```

Comment:
@xlc @zjb0807 This isn't making sense to me. `factory_contract_address` balance should `amount`. Alice is the one who should pay the deposit.

Reply:
For indirect contract creation via call instead create / create2, it is the contract paying the deposit, not the caller. Contract may require caller to send token for the contract deposit but it can also cover the fee for user.

Reply:
In that case `alice` balance should be `INITIAL_BALANCE - amount`, right?